src/execution_plan/execution_plan.c

```diff
@@ -498,304 +482,360 @@ static void _NewExecutionPlanSegment(RedisModuleCtx *ctx, GraphContext *gc,
 	}
 
 	AR_ExpNode **order_expressions = NULL;
-	if(order_clause) order_expressions = _BuildOrderExpressions(segment->record_map,
-																	segment->projections, order_clause);
+	if(order_clause) segment->order_expressions = _BuildOrderExpressions(segment->record_map,
+																			 segment->projections, order_clause);
+}
 
-	// Extend the RecordMap to include references from clauses that do not form projections
-	// (SET, CREATE, DELETE)
-	_ExecutionPlanSegment_MapReferences(segment, ast);
+static void _ExecutionPlanSegment_PlaceFilterOps(ExecutionPlanSegment *segment) {
+	Vector *sub_trees = FilterTree_SubTrees(segment->filter_tree);
+
+	/* For each filter tree find the earliest position along the execution
+	 * after which the filter tree can be applied. */
+	for(int i = 0; i < Vector_Size(sub_trees); i++) {
+		OpBase *op;
+		FT_FilterNode *tree;
+		Vector_Get(sub_trees, i, &tree);
+		rax *references = FilterTree_CollectModified(tree);
+
+		if(raxSize(references) > 0) {
+			/* Scan execution segment, locate the earliest position where all
+			 * references been resolved. */
+			op = ExecutionPlan_LocateReferences(segment->root, references);
+			assert(op);
+		} else {
+			/* The filter tree does not contain references, like:
```

Comment:
How about dropping these filters?

Reply:
A 1=1 filter can be safely dropped, but a 1=0 filter will cause no records to pass, so that would be a surprise!

Using a more practical example, a user might want to sample data with a filter like:
`MATCH (a) WHERE rand() < 0.1 RETURN a`
(Collect a random sampling of ~10% size of all nodes)

Reply:
👍